src/Mod/Path/PathScripts/PathDressupTrochoidal.py

```diff
@@ -0,0 +1,388 @@
+#  -*- coding: utf-8 -*-
+
+# ***************************************************************************
+# *                                                                         *
+# *   Copyright (c) 2018 LTS <SammelLothar@gmx.de> under LGPL               *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+from __future__ import print_function
+
+import FreeCAD
+import FreeCADGui
+import Path
+import Part
+import PathScripts.PathDressup as PathDressup
+import PathScripts.PathLog as PathLog
+import PathScripts.PathUtils as PathUtils
+import math
+import DraftVecUtils as D
+from PySide import QtCore
+from PathScripts.PathGeom import PathGeom
+
+"""Trochoidal Milling Path Dressup to profile"""
+
+# Qt tanslation handling
+def translate(text, context="Path_DressupTrochoidal", disambig=None):
+    return QtCore.QCoreApplication.translate(context, text, disambig)
+
+PathLog.setLevel(PathLog.Level.INFO, PathLog.thisModule())
+
+movecommands = ['G1', 'G01','G2', 'G3', 'G02', 'G03']
+rapidcommands = ['G0', 'G00']
+arccommands = ['G2', 'G3', 'G02', 'G03']
+global currLocation
+currLocation = {}
+
+class ObjectDressup:
+
+    def __init__(self, obj):
+        self.obj = obj
+        obj.addProperty("App::PropertyLink", "Base",  "Path", QtCore.QT_TRANSLATE_NOOP("App::Property", "The base path to modify"))
+        obj.addProperty("App::PropertyDistance", "StepWidth", "Path", QtCore.QT_TRANSLATE_NOOP("App::Property", "Step Alang Path"))
+        obj.addProperty("App::PropertyDistance", "MaterialAlowence", "Path", QtCore.QT_TRANSLATE_NOOP("App::Property", "Keep Material to Finish"))
```

Comment:
Should be Material**Allowance**
also 
is this correct: `"Keep Material to Finish"` ?